cannon/mipsevm/memory/memory.go

```diff
@@ -27,6 +27,20 @@ const (
 
 type Word = arch.Word
 
+type MappedMemoryRegion struct {
+	start_addr Word
+	end_addr   Word
```

Comment:
nit: camelCase for Go identifiers is the convention.
```suggestion
	startAddr Word
	endAddr   Word
```